File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int max = -1;
        int i=0, j=0, zeros = 0;
        while(j<nums.length) {
            if(nums[j] == 0)
                zeros++;

            j++;
            
            while(zeros > 1){
                if(nums[i] == 0)
                    zeros--;
                i++;
            }
            max = Math.max(j - i - 1, max);

        }
        return max;
    }
}

File: 1586-longest-subarray-of-1s-after-deleting-one-element/longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        start = end = 0
        zeroes = res = 0
        while end < len(nums):
            if nums[end] == 0:
                zeroes+=1
            end +=1

            if zeroes > 1:
                if nums[start] == 0:
                    zeroes -=1
                start+=1

            res = max(end - start - 1,res)
        return res